src/components/search/SearchDirective.js

```diff
@@ -149,40 +153,31 @@ goog.require('ga_search_type_directives');
             // Standard query then
             var url = gaUrlUtils.append($scope.options.searchUrl,
                                         'searchText=' + encodeURIComponent(q));
-            url = gaUrlUtils.append(url, 'lang=' + $translate.use());
-            url = $scope.options.applyTopicToUrl(url,
-                $scope.childoptions.currentTopic);
+            url = gaUrlUtils.append(url, 'lang=' + gaLang.get());
+            url = url.replace('{Topic}', gaTopic.get().id);
 
             $scope.childoptions.baseUrl = url;
             $scope.childoptions.query = q;
           }
         } else {
           blockQuery = false;
+          $scope.$broadcast('gaSearchReset');
         }
       };
 
-      $scope.trigger = function() {
-        startQuery($scope.query);
-      };
-
-      $scope.$watch('query', function(newval, oldval) {
-        $scope.trigger();
-      });
-
-      $scope.$on('gaTopicChange', function(evt, data) {
-        $scope.childoptions.currentTopic = data.id;
-      });
-
-      $scope.updateHref = function() {
-        $scope.encodedPermalinkHref = encodeURIComponent(gaPermalink.getHref());
-      };
-
-      $rootScope.$on('$translateChangeEnd', function() {
-        $scope.lang = $translate.use();
+      // We begin to watch the query only when topics are loaded
+      gaTopic.getTopics().then(function() {
+        $scope.topicLoaded = true;
+        if ($scope.query) {
```

Comment:
Is it even possible that 'query' contains something here? With swisssearch parameter?